orangecontrib/text/import_documents.py

```diff
@@ -185,10 +185,13 @@ class UrlReader(Reader, CoreUrlReader):
 
     def __init__(self, path, *args):
         CoreUrlReader.__init__(self, path)
-        Reader.__init__(self, self.filename, *args)
+        Reader.__init__(self, path, *args)
 
     def read_file(self):
-        self.filename = quote(self.filename, safe="/:")
+        # unquote prevent double quoting when filename is already quoted
+        # when not quoted it doesn't change url - it is required since Orange's
+        # UrlReader quote urls in version 3.29 but not in older versions
+        self.filename = quote(unquote(self.filename), safe="/:")
```

Comment:
Eventually this line could be removed. Should we write a test for it?

Reply:
I think this line cannot be removed if we support older versions of Orange (3.28) where quoting is not done by Orange's `Reader`. This is the reason why I decided on this solution. Anyway, we can also decide to raise the version of Orange to 3.29 and then this line would not be required.